File: spring-boot-cdc-event-streaming/spring-boot-cdc-event-streaming-mastodon-consumer/src/main/java/com/onlyteo/sandbox/config/RestConsumerConfig.java
package com.onlyteo.sandbox.config;

import jakarta.validation.Valid;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@EnableConfigurationProperties(AppRestProperties.class)
@Configuration(proxyBeanMethods = false)
public class RestConsumerConfig {

    @Bean
    public RestClient mastodonRestClient(@Valid final AppRestProperties properties) {
        return RestClient.builder()
                .baseUrl(properties.consumer().mastodon().url().toString())
                .defaultHeaders(headers -> headers.setBearerAuth(properties.consumer().mastodon().bearerToken()))
                .build();
    }
}

File: spring-boot-oauth2-react-client/spring-boot-oauth2-react-client-frontend-api/src/main/kotlin/com/onlyteo/sandbox/config/RestConsumerConfig.kt
package com.onlyteo.sandbox.config

import com.onlyteo.sandbox.properties.AppRestProperties
import jakarta.validation.Valid
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@EnableConfigurationProperties(AppRestProperties::class)
@Configuration(proxyBeanMethods = false)
class RestConsumerConfig {

    @Bean
    fun backendRestClient(@Valid properties: AppRestProperties): RestClient {
        return RestClient.builder()
            .baseUrl(properties.consumer.backend.url.toString())
            .build()
    }
}